impl/src/main/java/org/ehcache/impl/internal/store/heap/OnHeapStore.java

```diff
@@ -250,6 +251,12 @@ public OnHeapStore(final Configuration<K, V> config, final TimeSource timeSource
     getOrComputeIfAbsentObserver = operation(CachingTierOperationOutcomes.GetOrComputeIfAbsentOutcome.class).named("getOrComputeIfAbsent").of(this).tag("onheap-store").build();
     invalidateObserver = operation(CachingTierOperationOutcomes.InvalidateOutcome.class).named("invalidate").of(this).tag("onheap-store").build();
     silentInvalidateObserver = operation(HigherCachingTierOperationOutcomes.SilentInvalidateOutcome.class).named("silentInvalidate").of(this).tag("onheap-store").build();
+    StatisticsManager.createPassThroughStatistic(this, "size", Collections.singleton("onheap-store"), new Callable<Number>() {
+      @Override
+      public Number call() throws Exception {
+        return map.size();
```

Comment:
Indeed, but then I would name it differently, like `mappingsCount` or something else.